mlflow/types/schema.py

```diff
@@ -122,34 +127,401 @@ def from_numpy_type(cls, np_type):
+        self._check_properties(properties)
+        # Sort by name to make sure the order is stable
+        self._properties = sorted(properties)
+
+    def _check_properties(self, properties):
+        if not isinstance(properties, list):
+            raise MlflowException.invalid_parameter_value(
+                f"Expected properties to be a list, got type {type(properties).__name__}"
+            )
+        if len(properties) == 0:
+            raise MlflowException.invalid_parameter_value(
+                "Creating Object with empty properties is not allowed."
+            )
+        if any(not isinstance(v, Property) for v in properties):
+            raise MlflowException.invalid_parameter_value(
+                "Expected values to be instance of Property"
+            )
+        # check duplicated property names
+        names = [prop.name for prop in properties]
+        duplicates = {name for name in names if names.count(name) > 1}
+        if len(duplicates) > 0:
+            raise MlflowException.invalid_parameter_value(
+                f"Found duplicated property names: {duplicates}"
+            )
+
+    @property
+    def properties(self) -> List[Property]:
+        """The list of object properties"""
+        return self._properties
+
+    @properties.setter
+    def properties(self, value: List[Property]) -> None:
+        self._check_properties(value)
+        self._properties = sorted(value)
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, Object):
+            return self.properties == other.properties
+        return False
+
+    def to_dict(self):
+        properties = {
+            name: value for prop in self.properties for name, value in prop.to_dict().items()
+        }
+        return {
+            "type": OBJECT_TYPE,
+            "properties": properties,
+        }
+
+    @classmethod
+    def from_json_dict(cls, **kwargs):
+        """
+        Deserialize from a json loaded dictionary.
+        The dictionary is expected to contain `type` and
+        `properties` keys.
+        Example: {"type": "object", "properties": {"property_name": {"type": "string"}}}
+        """
+        if not {"properties", "type"} <= set(kwargs.keys()):
+            raise MlflowException(
+                "Missing keys in Object JSON. Expected to find keys `properties` and `type`"
+            )
+        if kwargs["type"] != OBJECT_TYPE:
+            raise MlflowException("Type mismatch, Object expects `object` as the type")
+        if not isinstance(kwargs["properties"], dict) or any(
+            not isinstance(prop, dict) for prop in kwargs["properties"].values()
+        ):
+            raise MlflowException("Expected properties to be a dictionary of Property JSON")
+        return cls(
+            [Property.from_json_dict(**{name: prop}) for name, prop in kwargs["properties"].items()]
+        )
+
+    def _merge_object(self, obj: "Object") -> "Object":
```

Comment:
```suggestion
    def _merge(self, obj: "Object") -> "Object":
```

Do we need `_object`?